pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -390,6 +424,67 @@ func handleZedrouterRestarted(ctxArg interface{}, restartCounter int) {
 	}
 }
 
+// handleAppInstanceStatusCreate - Handle AIS create. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusCreate(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+// handleAppInstanceStatusModify - Handle AIS modify. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}, oldStatusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+// handleAppInstanceStatusDelete - Handle AIS delete. Publish AppStatusSummary to ledmanager
+func handleAppInstanceStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+	ctx := ctxArg.(*zedmanagerContext)
+	publishAppInstanceSummary(ctx)
+}
+
+func publishAppInstanceSummary(ctxPtr *zedmanagerContext) {
+
+	summary := types.AppInstanceSummary{
+		TotalStarting: 0,
+		TotalRunning:  0,
+		TotalStopping: 0,
+		TotalError:    0,
+	}
+	items := ctxPtr.pubAppInstanceStatus.GetAll()
+	for _, st := range items {
+		status := st.(types.AppInstanceStatus)
+
+		if status.EffectiveActivate && status.Activated {
+			summary.TotalRunning++
+		} else {
+
+			switch status.State {
+			case types.INITIAL, types.DOWNLOADING, types.LOADING, types.CREATING_VOLUME, types.BOOTING:
```

Comment:
There is a much larger sets of states indicating that the app should be running. But instead of checking all the states here why just do your current
if status.EffectiveActivate && status.Activated {
TotalRunning++
} else if len(status.Error) > 0 {
TotalError++
} else if status.Activated {
TotalStopping++
} else { // !status.EffectiveActivate 
TotalStarting++
}

Reply:
I agree looks simple, will try that

Reply:
There is one issue in the logic above which I found during my testing. When user explicitly halted the app the status is set as follows, Activated = false and EffectiveActive = false which is correct and we should not even light any led.
But with logic above we consider that state as Starting and blinking green.

Modified that as

if status.EffectiveActivate && status.Activated {
TotalRunning++
} else if len(status.Error) > 0 {
TotalError++
} else if status.Activated {
TotalStopping++
} else if status.EffectiveActive {
TotalStarting++
}